libvirt/resource_libvirt_domain.go

```diff
@@ -256,6 +256,9 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		domainDef.Devices.Console = consoles
 	}
 
+	controller := defController{Type:"scsi", Model:"virtio-scsi"}
+	domainDef.Devices.Controller = append(domainDef.Devices.Controller, controller)
```

Comment:
The controller is still there but if the disk isn't scsi it won't be used.  As far as I can tell libvirt doesn't remove the controller definition if it isn't used.